src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/StreamPipeReader.cs

```diff
@@ -383,18 +383,21 @@ public override async Task CopyToAsync(PipeWriter destination, CancellationToken
                 try
                 {
                     BufferSegment? segment = _readHead;
+                    int segmentIndex = _readIndex;
+
                     try
                     {
                         while (segment != null)
                         {
-                            FlushResult flushResult = await destination.WriteAsync(segment.Memory, tokenSource.Token).ConfigureAwait(false);
+                            FlushResult flushResult = await destination.WriteAsync(segmentIndex > 0 ? segment.Memory.Slice(segmentIndex) : segment.Memory, tokenSource.Token).ConfigureAwait(false);
```

Comment:
Would it make sense to just always slice rather than doing the conditional?

Reply:
Probably, it did feel like a premature optimization